tests/thread-safety/source.ml

```diff
@@ -0,0 +1,13 @@
+let () =
+  let rec repeat n =
+    if n <= 0 then ()
+    else repeat (n - 1)
+  in
```

Comment:
The threads are interleaving, just not in the middle of updating a count. I would have to insert the `Thread.yield` call into the instrumentation, not into the code being instrumented.